internal/mutators/namespace.go

```diff
@@ -52,19 +53,66 @@ func (m *Namespace) Handle(ctx context.Context, req admission.Request) admission
 // Currently, we only add `included-namespace` label to non-excluded namespaces
 // if the label is missing.
 func (m *Namespace) handle(log logr.Logger, ns *corev1.Namespace) {
-	// Early exit if the namespace is excluded.
-	if config.ExcludedNamespaces[ns.Name] {
-		return
-	}
 
-	// Add label if the namespace doesn't have it.
-	if _, hasLabel := ns.Labels[api.LabelIncludedNamespace]; !hasLabel {
-		if ns.Labels == nil {
-			ns.Labels = map[string]string{}
+	// if includedNamespaces is set, we dont need excluded as everyone that is not included is
+	// automatically excluded.
+	// if includedNamespaces is set, we need to set the label only to the included namespaces list
+
+	// if parameter included-namespaces is set, everything else is excluded
+	// so, we set the annotation only if ns exists in the map
+
+	//log.Info("we are in the mutator handle and this is the excluded map:", config.ExcludedNamespaces)
+
+	if config.IncludedNamespaces != nil {
```

Comment:
if --include-namespaces is set, the logic is that everything else is excluded.
This code is repited in some other parts ( mutators also ) , so I need to extract that in a separate function.
Dont know where to place this function.